src/lib/annotations/annotations.html

```diff
@@ -88,26 +88,77 @@
           this._parseElementAnnotations(node, list);
     },
 
-    _testEscape: function(value) {
-      var escape = value.slice(0, 2);
-      if (escape === '{{' || escape === '[[') {
-        return escape;
+    _bindingRegex: /([^{[]*)({{|\[\[)([^}\]]*)(?:]]|}})/g,
+
+    _parseBindings: function(text) {
+      var re = this._bindingRegex;
+      var parts = [];
+      var m, lastIndex;
+      // Example: "literal1{{binding1}}literal2[[binding2]]final"
+      // Regex matches:
+      //       Iteration 1:  Iteration 2:
+      // m[1]: 'literal1'    'literal2'
+      // m[2]: '{{'          '[['
+      // m[3]: 'binding1'    'binding2'
+      // 'final' is manually substring'ed from end
+      while ((m = re.exec(text)) !== null) {
+        // Add literal part
+        if (m[1]) {
+          parts.push({literal: m[1]});
+        }
+        // Add binding part
+        // Mode (one-way or two)
+        var mode = m[2][0];
+        var value = m[3].trim();
+        // Negate
+        var negate = false;
+        if (value[0] == '!') {
+          negate = true;
+          value = value.substring(1);

```

Comment:
Since you're accepting whitespace around the value (L112), should this be `value.substring(1).trim()` for consistency? i.e., should we accept:

    {{! foo}}

as well as:

    {{ !foo }}